File: articles/energy-work-and-conservation-in-physical-systems/fortran/spring_energy_table.f90
! Spring Energy Table
!
! This Fortran workflow computes:
!
!   U = 1/2 k x^2
!   v = x sqrt(k/m)

program spring_energy_table
  implicit none

  integer :: i
  real(8), parameter :: mass_kg = 0.50d0
  real(8), parameter :: spring_constant_n_per_m = 20.0d0
  real(8), dimension(6) :: compressions_m
  real(8) :: x
  real(8) :: spring_energy_j
  real(8) :: predicted_speed_m_per_s

  compressions_m = (/0.02d0, 0.04d0, 0.06d0, 0.08d0, 0.10d0, 0.12d0/)

  print *, "mass_kg,spring_constant_n_per_m,compression_m,spring_energy_j,predicted_speed_m_per_s"

  do i = 1, size(compressions_m)
     x = compressions_m(i)
     spring_energy_j = 0.5d0 * spring_constant_n_per_m * x**2
     predicted_speed_m_per_s = x * sqrt(spring_constant_n_per_m / mass_kg)

     print '(F10.5,A,F10.5,A,F10.5,A,ES14.6,A,F14.8)', &
       mass_kg, ",", spring_constant_n_per_m, ",", x, ",", spring_energy_j, ",", predicted_speed_m_per_s
  end do

end program spring_energy_table

File: articles/energy-work-and-conservation-in-physical-systems/c/spring_energy_table.c
/*
Spring Energy Table

This C workflow computes:

    U = 1/2 k x^2
    v = x sqrt(k/m)

for a spring-launch model.
*/

#include <math.h>
#include <stdio.h>

int main(void) {
    const double mass_kg = 0.50;
    const double spring_constant_n_per_m = 20.0;

    double compressions_m[] = {0.02, 0.04, 0.06, 0.08, 0.10, 0.12};
    int n = sizeof(compressions_m) / sizeof(compressions_m[0]);

    printf("mass_kg,spring_constant_n_per_m,compression_m,spring_energy_j,predicted_speed_m_per_s\n");

    for (int i = 0; i < n; i++) {
        double x = compressions_m[i];
        double spring_energy_j = 0.5 * spring_constant_n_per_m * x * x;
        double predicted_speed_m_per_s = x * sqrt(spring_constant_n_per_m / mass_kg);

        printf("%.6f,%.6f,%.6f,%.8f,%.8f\n",
               mass_kg,
               spring_constant_n_per_m,
               x,
               spring_energy_j,
               predicted_speed_m_per_s);
    }

    return 0;
}
